heron/api/src/java/org/apache/heron/streamlet/impl/streamlets/GeneralReduceByKeyAndWindowStreamlet.java

```diff
@@ -66,7 +65,7 @@ public boolean doBuild(TopologyBuilder bldr, Set<String> stageNames) {
     setDefaultNameIfNone(StreamletNamePrefix.REDUCE, stageNames);
     GeneralReduceByKeyAndWindowOperator<K, V, VR> bolt =
         new GeneralReduceByKeyAndWindowOperator<K, V, VR>(keyExtractor, identity, reduceFn);
-    windowCfg.attachWindowConfig(bolt);
+    bolt.applyWindowConfig(windowCfg);
```

Comment:
`applyWindowConfig` and `attachWindowConfig` seem redundant to each other. We could call `windowCfg.attachWIndowConfig(bolt)` here directly right?

Reply:
windowCfg doesn't have the function and requires a static cast. :(

Reply:
Cleaned up a bit more. Moved the call into Operator classes so we dont need to worry about the details here.

Reply:
> windowCfg doesn't have the function and requires a static cast. :(

hmm, if we combine `WindowConfig` and `WindowConfigImpl` into one class as an abstract class, then `windowCfg.attachWindowConfig(bolt)` should work.

Reply:
Yeah. Thanks for figuring out the cause of the compiling error when the attachWindowConfig() is in the interface.

Let me update it.